site/app/controllers/admin/ReportController.php

```diff
@@ -666,62 +704,174 @@ public function generateCustomization() {
             $this->core->addErrorMessage($msg);
+            $redirect_url = $this->core->buildCourseUrl(['reports', 'rainbow_grades_customization']);
             return new MultiResponse(
                 JsonResponse::getErrorResponse($msg),
                 null,
                 new RedirectResponse($redirect_url)
             );
         }
+    }
 
-        $msg = 'Rainbow Grades Customization uploaded';
-        $this->core->addSuccessMessage($msg);
-        return new MultiResponse(
-            JsonResponse::getSuccessResponse([
-                'customization_path' => $rainbow_grades_dir
-            ]),
-            null,
-            new RedirectResponse($redirect_url)
-        );
+
+    #[Route('/courses/{_semester}/{_course}/reports/rainbow_grades_customization/manual_or_gui', methods: ['POST'])]
+    public function setRainbowGradeCustomization(): JsonResponse {
+
+        // Extract the value from $_POST
+        $selectedValue = $_POST['selected_value'] ?? null;
+
+        if ($selectedValue === null || trim($selectedValue) === '') {
+            $msg = 'Invalid request: No selected value provided.';
+            return JsonResponse::getErrorResponse($msg);
+        }
+
+        $rainbow_grades_dir = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "rainbow_grades");
+        $customization_dest = FileUtils::joinPaths($rainbow_grades_dir, 'customization.json');
+
+        // Determine the source file based on the selected value
+        switch ($selectedValue) {
+            case 'manual':
+                $customization_src = FileUtils::joinPaths($rainbow_grades_dir, 'manual_customization.json');
+                break;
+            case 'gui':
+                $customization_src = FileUtils::joinPaths($rainbow_grades_dir, 'gui_customization.json');
+                break;
+            default:
+                $msg = 'Invalid request: Unknown selected value.';
+                return JsonResponse::getErrorResponse($msg);
+        }
+
+        // Copy the source file to the destination
+        if (!copy($customization_src, $customization_dest)) {
+            $msg = 'File copy failed: Could not copy file.';
+            return JsonResponse::getErrorResponse($msg);
+        }
+
+        $msg = 'Rainbow Grades Customization set successfully';
+
+        return JsonResponse::getSuccessResponse([
+            'selected_value' => $selectedValue,
+            'message' => $msg,
+        ]);
     }
 
-    #[Route("/courses/{_semester}/{_course}/reports/rainbow_grades_status")]
+
+
+    #[Route('/courses/{_semester}/{_course}/reports/rainbow_grades_status', methods: ['POST'])]
     public function autoRainbowGradesStatus() {
         // Create path to the file we expect to find in the jobs queue
         $jobs_file = '/var/local/submitty/daemon_job_queue/auto_rainbow_' .
             $this->core->getConfig()->getTerm() .
             '_' .
             $this->core->getConfig()->getCourse() .
             '.json';
-
```

Comment:
This whitespace change doesn't appear to be related to the rest of the PR and should be reverted.